src/main/java/FileNavigator.java

```diff
@@ -0,0 +1,65 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Class for file navigator
+ */
+public class FileNavigator {
+    /**
+     * Files
+     */
+    public HashMap<String, HashSet<FileData>> Files = new HashMap<>();
+
+    /**
+     * Add file data
+     *
+     * @param fileData file data
+     */
+    public void add(FileData fileData) {
+        if (fileData == null)
+            throw new NullPointerException("File data is null");
+        if (fileData.Name == null)
+            throw new NullPointerException("File name is null");
+        if (fileData.Path == null)
+            throw new NullPointerException("File path is null");
+        if (fileData.Size <= 0)
+            throw new IllegalArgumentException("File size is less or equal to zero");
+        if (Files.containsKey(fileData.Path)) {
+            Set<FileData> fileDataSet = Files.get(fileData.Path);
+            if (fileDataSet.contains(fileData))
```

Comment:
Contains comparing in this case only links to objects. You don't write in FileData.class `equals` and `hashCode`.

You may have two FileData instances with the same information (one already in `fileDataSet`), but `fileDataSet.contains(fileData)` will return false